### ISSUE TITLE
Feature request: make `target=_blank rel=…` for external links optional

### ISSUE THREAD

**Issue by mathiasbynens** [vuejs/vuepress#186]
https://github.com/vuejs/vuepress/blob/4cf123240322898c44cab0ada5a729ca66ba9094/lib/markdown/link.js#L16

`target=_blank` denies people the opportunity to open the link in the same window. Please consider making it possible to disable the automatic addition of these attributes.

**Comment by lbenie** [comment]
would it be best to add an option to the base config ?

**Comment by ulivz** [comment]
@mathiasbynens 

We can support following format to add custom attributes for external link:

```md
[link-text]((querystring)link)
```

For example:

```md
[custom domain]((target=_self&class=my-link)http://surge.sh/)
```

It will be rendered as:

```
<a href="http://surge.sh/" target="_self" class="my-link">surge</a>
```

cc @yyx990803

**Comment by ycmjason** [comment]
I have created two PR with two approaches. Perhaps we need to decide which way we like more. I personally like #357 more as it is less low-level.

**Comment by ycmjason** [comment]
@ulivz please close this as #358 is merged.

**Comment by ulivz** [comment]
Fixed at 20e5bd89540d13af9426967eae84e30a97c7b65b, thank for @ycmjason
